front/components/vaults/EditVaultStaticDatasourcesViews.tsx

```diff
@@ -0,0 +1,88 @@
+import { Button, PlusIcon, Popup } from "@dust-tt/sparkle";
+import type {
+  DataSourceType,
+  PlanType,
+  VaultType,
+  WorkspaceType,
+} from "@dust-tt/types";
+import { useRouter } from "next/router";
+import React, { useState } from "react";
+
+import VaultCreateFolderModal from "@app/components/vaults/VaultCreateFolderModal";
+import VaultCreateWebsiteModal from "@app/components/vaults/VaultCreateWebsiteModal";
+import { useSubmitFunction } from "@app/lib/client/utils";
+
+export function EditVaultStaticDataSourcesViews({
+  owner,
+  plan,
+  vault,
+  category,
+  dataSources,
+}: {
+  owner: WorkspaceType;
+  plan: PlanType;
+  vault: VaultType;
+  category: "folder" | "website";
+  dataSources: DataSourceType[];
+}) {
+  const router = useRouter();
+  const [showAddFolderModal, setShowAddFolderModal] = useState(false);
+  const [showAddWebsiteModal, setShowAddWebsiteModal] = useState(false);
+  const [showDatasourceLimitPopup, setShowDatasourceLimitPopup] =
+    useState(false);
+
+  const planDataSourcesLimit = plan.limits.dataSources.count;
+  const { submit: checkLimitsAndOpenModal } = useSubmitFunction(async () => {
```

Comment:
We should not use `useSubmitFunction` for synchronous case as it bundles some "loading state" logic. You can do a regular useCallback here.